From 10ce380a196d3362689b54ac294666cec3ed36d2 Mon Sep 17 00:00:00 2001
From: David Fries <David@Fries.net>
Date: Sun, 26 Aug 2012 16:42:18 -0500
Subject: [PATCH] disable getnstimeofday WARN_ON

pm_dbg_update_time is calling getnstimeofday, for the N900
omap_sram_idle transitions to and from suspend and calls routines
which call pm_dbg_update_time.  Calling
timekeeping_suspend/timekeeping_resume doesn't seem very doable.
Disabling it seems like the easier option.
---
 kernel/time/timekeeping.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/time/timekeeping.c b/kernel/time/timekeeping.c
index 900f1b6..6d81608 100644
--- a/kernel/time/timekeeping.c
+++ b/kernel/time/timekeeping.c
@@ -95,7 +95,12 @@ void getnstimeofday(struct timespec *ts)
 	unsigned long seq;
 	s64 nsecs;
 
+	/* pm_dbg_update_time calls getnstimeofday when timekeeping is
+	 * suspended, disable it
+	 */
+	/*
 	WARN_ON(timekeeping_suspended);
+	*/
 
 	do {
 		seq = read_seqbegin(&xtime_lock);
-- 
1.7.10.4

